developing/structure/transaction.md

```diff
@@ -0,0 +1,1152 @@
+            },
+            ...
+          ]
+        },
+        {
+          "type": "message",
+          "attributes": [
+            {
+              "key": "action",
+              "value": "/osmosis.gamm.v1beta1.MsgSwapExactAmountIn"
+            },
+            {
+              "key": "sender",
+              "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+            },
+            {
+              "key": "sender",
+              "value": "osmo1thscstwxp87g0ygh7le3h92f9ff4sel9y9d2eysa25p43yf43rysk7jp93"
+            },
+            ...
+          ]
+        },
+        {
+          "type": "token_swapped",
+          "attributes": [
+            {
+              "key": "module",
+              "value": "gamm"
+            },
+            {
+              "key": "sender",
+              "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+            },
+            {
+              "key": "pool_id",
+              "value": "604"
+            },
+            {
+              "key": "tokens_in",
+              "value": "15000000uosmo"
+            },
+            {
+              "key": "tokens_out",
+              "value": "289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4"
+            },
+            {
+              "key": "module",
+              "value": "gamm"
+            },
+            {
+              "key": "sender",
+              "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+            },
+            {
+              "key": "pool_id",
+              "value": "611"
+            },
+            {
+              "key": "tokens_in",
+              "value": "289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4"
+            },
+            {
+              "key": "tokens_out",
+              "value": "3802225ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
+            },
+            ...
+          ]
+        },
+        {
+          "type": "transfer",
+          "attributes": [
+            {
+              "key": "recipient",
+              "value": "osmo1thscstwxp87g0ygh7le3h92f9ff4sel9y9d2eysa25p43yf43rysk7jp93"
+            },
+            {
+              "key": "sender",
+              "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+            },
+            {
+              "key": "amount",
+              "value": "15000000uosmo"
+            },
+            {
+              "key": "recipient",
+              "value": "osmo1l4u56l7cvx8n0n6c7w650k02vz67qudjlcut89"
+            },
+            {
+              "key": "sender",
+              "value": "osmo1thscstwxp87g0ygh7le3h92f9ff4sel9y9d2eysa25p43yf43rysk7jp93"
+            },
+            {
+              "key": "amount",
+              "value": "289236271ibc/987C17B11ABC2B20019178ACE62929FE9840202CE79498E29FE8E5CB02B7C0A4"
+            },
+            ...
+          ]
+        }
+      ]
+    }
+  ],
```

Comment:
I think some of these code blocks should maybe be hidden by a dropdown like so
::: details Example
:::

An example of this is here https://docs.osmosis.zone/developing/modules/spec-mint.html#params

Just a suggestion to make it easier to read IMO

Reply:
Oh I really like that. Going to open a new PR with some of these, including this.